stratospheric-dev/stratospheric
# Use the cdk.json file to define configuration values for our CDK stacks

## Why?

Bootstrapping the entire Stratospheric infrastructure in a reader's AWS account is currently not straightforward (see https://github.com/stratospheric-dev/stratospheric/issues/64).

With this PR, we're leveraging the `cdk.json` file to store configuration values at a central place. This makes it easy to adjust the configuration for a new AWS account. 

I'm planning to adjust the relevant sections in our eBook once this PR is approved. Tom will adjust the initial chapter on CDK to explain this approach.

## How to verify?

To verify the main infrastructure stacks for our application, try to replicate the steps as presented in `cdk/STEPS.md` (-> now part of the appendix of the book), preferably with a new domain, or use our Stratospheric AWS account.

Creating the entire infrastructure from scratch takes 20-30 minutes.

There shouldn't be any hardcoded parameters (`-c value=foo`) inside our `package.json` files anymore

Afterward, make sure you're able to register an account and create a todo item.

After a successful deployment and verification, make sure to destroy the entire infrastructure again.

## Additional changes

- apply this new way of passing the parameters for the chapter versions
- simplify and unify our GitHub workflows to avoid duplication
- update to `cdk` 1.124.0 for the chapter versions
- define node `>=16` for chapter versions
